network/src/peer_manager/mod.rs

```diff
@@ -816,24 +829,44 @@ where
     ) {
         let mut upstream_handlers = self.upstream_handlers.clone();
         let network_context = self.network_context.clone();
+        let inbound_rate_limiter = self.inbound_rate_limiter.clone();
         self.executor.spawn(network_events.for_each_concurrent(
             self.max_concurrent_network_reqs,
             move |inbound_event| {
                 Self::handle_inbound_event(
                     network_context.clone(),
                     inbound_event,
                     peer_id,
+                    inbound_rate_limiter.clone(),
                     &mut upstream_handlers,
                 );
                 futures::future::ready(())
             },
         ));
     }
 
+    fn rate_limit<'a>(
+        inbound_rate_limiter: &'a RateLimiter<PeerId, DefaultKeyedStateStore<PeerId>, DefaultClock>,
+        peer_id: &PeerId,
+        msg_length: usize,
+    ) -> Result<(), NegativeMultiDecision<NotUntil<'a, QuantaInstant>>> {
+        // We can't rate limit something of 0 size
+        if msg_length < 1 {
+            return Ok(());
+        }
```

Comment:
what does it matter?

Reply:
This is because Governor takes a `NonZeroU32`.  So, if it was of size 0, it would fail on the result down below.

The reason for all of this mess was because I was working around a returned lifetime value that coincides with the `inbound_rate_limiter` lifetime in itself.  So, after a lot of iteration I got here just to get this function to work so I wouldn't have to repeat this block twice.